daemon/SecretsImpl/secretsrequestprocessor.cpp

```diff
@@ -1009,20 +1019,25 @@ Daemon::ApiImpl::RequestProcessor::setCollectionSecretGetAuthenticationCode(
 
         // perform the user input flow required to get the input key data which will be used
         // to unlock this collection.
-        InteractionParameters ikdRequest;
-        ikdRequest.setApplicationId(callerApplicationId);
-        ikdRequest.setPluginName(secret.identifier().storagePluginName());
-        ikdRequest.setCollectionName(secret.identifier().collectionName());
-        ikdRequest.setSecretName(secret.identifier().name());
-        ikdRequest.setOperation(InteractionParameters::StoreSecret);
-        ikdRequest.setInputType(InteractionParameters::AlphaNumericInput);
-        ikdRequest.setEchoMode(InteractionParameters::PasswordEchoOnEdit);
-        ikdRequest.setPromptText(tr("Enter the passphrase to unlock the collection %1 in order to store secret %2")
-                                 .arg(secret.identifier().collectionName(), secret.identifier().name()));
+        InteractionParameters promptParams;
+        promptParams.setApplicationId(callerApplicationId);
+        promptParams.setPluginName(secret.identifier().storagePluginName());
+        promptParams.setCollectionName(secret.identifier().collectionName());
+        promptParams.setSecretName(secret.identifier().name());
+        promptParams.setOperation(InteractionParameters::StoreSecret);
+        promptParams.setInputType(InteractionParameters::AlphaNumericInput);
+        promptParams.setEchoMode(InteractionParameters::PasswordEchoOnEdit);
+        //: This will be displayed to the user, prompting them to enter the passphrase to unlock the collection in which a new secret will be stored. %1 is the application name, %2 is the  secret name, %3 is the collection name, %4 is the plugin name.
+        //% "App %1 wants to store a new secret named %2 into collection %3 in plugin %4. Enter the passphrase to unlock the collection."
```

Comment:
App %1 -> %1 App

collection %3 -> %3 collection

Like said I would drop "App" when/if app icon can be fetched and shown inside the dialog (would then require two variants of each string).

"plugin" is null-word and quite technical term, I would drop it and just display the name of the plugin.

Reply:
So "App %1 wants to store a new secret named %2 into collection %3 in plugin %4." -> "%1 App wants to store a new secret, named %2, into %3 collection from %4."

Reply:
Why reorder that way?  To me "collection X" sounds more natural than "X collection", similarly "App Y" sounds more natural than "Y app".

For the icon stuff, can we postpone that because I'm not sure what we support in the dialogs yet, needs more investigation and potentially dbus API changes etc.

Reply:
Yeah let's postpone the icon support.